locator/topology.cc

```diff
@@ -580,6 +581,9 @@ void topology::sort_by_proximity(locator::host_id address, host_id_vector_replic
 
 template <typename T>
 void topology::do_sort_by_proximity(T address, utils::small_vector<T, 3>& addresses) const {
+    static thread_local std::mt19937_64 random_engine(std::random_device{}());
```

Comment:
Do we need some repeatable seed thing here for tests?

Reply:
Can consider a simpler random engine like linear congruential (may or may not help).

Reply:
There are others (xoshiro's) that are faster with smaller state size, I believe.

Reply:
I'm not sure it will make any difference, but we can have the random_engine live in topology
and expose a private method for the test to seed it.